source/lac/petsc_vector_base.cc

```diff
@@ -284,7 +284,10 @@ namespace PETScWrappers
         ghost_indices.set_size(this->size());
 
         if (std::is_sorted(&array[end_index - ghost_start_index],
-                           &array[n_elements_stored_locally]))
+                           &array[n_elements_stored_locally],
+                           [](PetscScalar left, PetscScalar right) {
+                             return std::real(left) < std::real(right);
```

Comment:
It may be clearest if you wrote this as follows:
```suggestion
                             return static_cast<PetscInt>(std::real(left)) < static_cast<PetscInt>(std::real(right));
```